### DISCUSSION TITLE
`-balance_levels`

### DISCUSSION THREAD

**Discussion by kareefardi** [The-OpenROAD-Project/OpenROAD#3955] [Q&A] [answered]
`-balance_levels` is not documented. What is does this flag do and what are the uses of it?
Edit: 
More context: It is a flag for `clock_tree_synthesis` command

**Comment by vvbandeira** [comment]
@luarss
did we miss this flag?

**Comment by maliberty** [comment, marked as answer]
When you have a non-register in the clock tree (eg clock-gate or inverter) then it attempts to ensure the are similar number of levels in the tree across that object.  The idea is captured in this image
![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/761514/eddb3913-086a-4c3f-8744-7a3cadc90543)
